hackathon/models.py

```diff
@@ -0,0 +1,174 @@
+from django.db import models
+from django.contrib.auth.models import User
+
+# Optional fields are ony set to deal with object deletion issues.
+# If this isn't a problem, they can all be changed to required fields.
+
+# The "updated" field isn't editable in admin. Neither is "submission" field.
+# A custom method to auto-fill created_by fields can be made
+
+# When it seemed relevant, I moved the ForeignKey field to another model and
+# used the one listed in the schema as the related_name. I may have been
+# erroneous to do so but it can be corrected easily.
+# My reasoning is explained in comments above the ForeignKey fields.
+
+
+class Hackathon(models.Model):
+    """Model representing a Hackathon. It is connected by a foreign key to 
+    User, HackAwards and HackTeam. Optional Fields: judges, organiser.
+    "awards" and "teams" are related tables. They have been moved to
+    HackAwardCategory and HackTeam respectively. Please see comments there."""
+    created = models.DateTimeField(auto_now_add=True)
+    updated = models.DateTimeField(auto_now=True)
+    # Each model can only be created by one user: One To Many
+    created_by = models.ForeignKey(User,
+                                   on_delete=models.CASCADE,
+                                   related_name="hackathon_created_by")
```

Comment:
Sorry, I should have mentioned this in the other comment, but I would just rename all the `related_name`s for  `created_by` to be just the plural of the model name in lower case. I think that would be the best seeing as it they are basically a collection of that model created by a user.

Reply:
Okay, makes sense, I'll do that now.